jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -726,10 +757,22 @@ private String setupGlobalSettingFile(@Nonnull Collection<Credentials> credentia
         }
 
         // Settings provided by the global maven configuration
-        GlobalSettingsProvider globalSettingsProvider = GlobalMavenConfig.get().getGlobalSettingsProvider();
+        GlobalSettingsProvider globalSettingsProvider;
+
+        MavenConfigFolderOverrideProperty overrideProperty = getMavenConfigOverrideProperty();
+        if (overrideProperty != null) {
+            // Settings overriden by a folder property
+            console.format("[withMaven] using overriden Maven global settings by folder '%s'%n", overrideProperty.getOwner().getDisplayName());

```

Comment:
Can we output this message only if `Level.FINE` is enabled?
See  [JENKINS-48657 - Noisy build log message "\[withMaven\] Skip 'xxx' disabled by configuration"](https://issues.jenkins-ci.org/browse/JENKINS-48657)